### PR TITLE
feat(Variance): `css` - Standard CSS Property Passthrough + Selector System Props [WEB-1117]

### PR DESCRIPTION
**Pull request by codecaaron**
### [Variance Migration Doc](https://www.notion.so/codecademy/Variance-Migration-a675bbc31b584fec9dfff72c0364b194)

Final Markdown documentation for these behaviors will be ticketed once the API has been finalized, as part of the cut over from`gamut-system`.

#### Static CSS Utility Types
```tsx
export type SelectorMap<Props, System> = {
  [K in keyof Props]?: SelectorProps<Props[K], System>;
};

export type SelectorProps<Props, System> = {
  [K in keyof Props]?: K extends keyof System
    ? System[K] // System Property Types
    : K extends keyof PropertyTypes
    ? PropertyTypes[K] // Any Valid CSS Prop
    : Omit<PropertyTypes, keyof System> & System // No Match Opens a single level of nesting for System + General CSS;
};
```

#### Accept and Pass Through all valid CSS properties on Static Utilities (as they are not props).
This change lets us configure anything that is likely to be non dynamic but required for a complete and discrete style.   The typings for both `variant` and `css` accept all properties that are not present on the passed configuration.

**These are still guaranteed to be typesafe by CSSType.**

These props can only be the value specified and will not accept values with the responsive syntax.  
```tsx
const styled.ul(variant({
   key: 'listVariant',
   variants: {
      vertical: {
         listStyle: 'none',  // not a system prop but valid 
         p: 0,
         m: 0,
      },
      emoji: {
         listStyleType: "\1F44D",  // not 
         pr: [12, 24],
         m: 0
      }
}))
```
When generating the styles the function will ignore the all props not specified by our configuration and simply pass them through to the result CSSObject. 

#### Selectors for Static Utilities
This change lets us specify scoped selector syntax for anything not matching a valid CSS prop or system property.  The typings here are very loose as the evaluation of possible selectors is very taxing even with stronger aliasing.   This with the addition of the CSS Passthrough will allows us to create complete variants for much more complex css as scale.  And create robust sharable utilities. 

```tsx
const styled.a(variant({
   variants: {
      interface: {
         color: 'navy',
         '&:hover': { 
             textDecoration: 'none',
             border: '2px solid currentColor',
             borderWidth: {  md: '4px' } // Responsive syntax party town
         }
      },
      inline: {
         color: 'hyper',
         transition: 'scale 150ms ease',
         '&:hover': {
            textDecoration: 'underline',
            transform: 'scale(1.2)', // static properties work here too!
         }
      }
}))
```

Immediate uses post migration: 
- **Anchor**
- **Button**
- **OrderedList / UnorderList**

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1117
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/variance@0.2.1-alpha.45b173.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6063a72e29ee7f00a3781bad--gamut-preview.netlify.app>https://6063a72e29ee7f00a3781bad--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6063a72e29ee7f00a3781bad>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/variance@0.2.2-alpha.83bf18.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6063ad1c1c9085009c29fc19--gamut-preview.netlify.app>https://6063ad1c1c9085009c29fc19--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6063ad1c1c9085009c29fc19>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/variance@0.2.2-alpha.c92529.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60640a14bb658f843802f648--gamut-preview.netlify.app>https://60640a14bb658f843802f648--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60640a14bb658f843802f648>Deploy Logs</a></p>

File: packages/variance/src/core.ts [hunk 1]
```diff
@@ -136,15 +137,40 @@ export const variance = {
         P extends Parser<T, TransformerMap<T, Config>>
       >(config: Config): CSS<T, P> {
         const parser = this.create(config);
-
+        const filteredProps: string[] = [...parser.propNames];
```
Comment (JoshuaKGoldberg) [inline packages/variance/src/core.ts:140]: Does this need to be copied? Looks to me like it's never modified?

**Review by JoshuaKGoldberg — APPROVED** [review]
Very nifty.

**Pull request merged**